src/lib.rs

```diff
@@ -6,23 +6,39 @@ pub(crate) mod signing_state_machine;
 pub(crate) mod utils;
 
 const META_SALT: &str = "wsts-protocol";
+
+#[macro_export]
+macro_rules! compute_sha256_hash {
+    ($($data:expr),*) => {
+        {
+            use k256::sha2::{Digest, Sha256};
+            let mut hasher = Sha256::default();
+            $(hasher.update($data);)*
+            let result = hasher.finalize();
+            let mut hash = [0u8; 32];
+            hash.copy_from_slice(result.as_slice());
+            hash
+        }
+    };
+}
+
 /// Helper function to compute deterministic hashes for the WSTS processes.
 /// Note: for signing, the "call_id" should be the call_id of the preceeding
 /// keygen job
-pub fn compute_deterministic_hashes(
+pub fn compute_execution_hashes(
```

Comment:
Remove this fn and just use gadget-crypto-hasing and just hash these values in place